src/box/relay.cc

```diff
@@ -883,13 +901,15 @@ relay_subscribe_f(va_list ap)
 
 		fiber_cond_wait_deadline(&relay->reader_cond,
 					 relay->last_row_time + timeout);
-
 		/*
 		 * The fiber can be woken by IO cancel, by a timeout of
 		 * status messaging or by an acknowledge to status message.
 		 * Handle cbus messages first.
 		 */
-		cbus_process(&relay->endpoint);
+		inj = errinj(ERRINJ_RELAY_FROM_TX_DELAY, ERRINJ_BOOL);
+		if (inj == NULL || !inj->bparam)
```

Comment:
I think there shouldn't be such places, where something happens if errinj is NOT set.
We usually add some actions when errinj IS set.
So I'd change this to:
```c
         ERROR_INJECT(ERRINJ_RELAY_FROM_TX_DELAY, {
                  goto skip_tx;
         });
         cbus_process(&relay->tx_endpoint);
skip_tx:
         cbus_process(&relay->wal_endpoint);
```

This is subjective, and the behaviour won't change with my suggestion. So, up to you.

Reply:
Above code doesn't build with Release build type because of

it can be changed to something like:
```c
    inj = errinj(ERRINJ_RELAY_FROM_TX_DELAY, ERRINJ_BOOL);
    if (inj != NULL && inj->bparam)
        goto skip_tx_process;
    cbus_process(&relay->tx_endpoint);
skip_tx_process:
    cbus_process(&relay->wal_endpoint);
```
But current version looks cleaner, so I'll leave it as is.

Reply:
There is a hack how to deal with that issue - just add `;` right after the label:
```C
skip_tx_process:;
```
Just a hint for the future. The current code for me looks fine.

Reply:
Thanks for the hack, might be useful someday!